src/Events/Category_Colors/Event_Category_Meta.php

```diff
@@ -0,0 +1,258 @@
+<?php
+/**
+ * Event_Category_Meta class for taxonomy meta.
+ * Handles metadata for terms within the `tribe_events_cat` taxonomy.
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\Category_Colors
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Events\Category_Colors;
+
+use InvalidArgumentException;
+use Tribe__Events__Main;
+use WP_Term;
+use WP_Error;
+
+/**
+ * Class Event_Category_Meta
+ * Provides an object-oriented way to set, retrieve, and delete metadata
+ * associated with event categories. It ensures that only valid terms within
+ * the `tribe_events_cat` taxonomy can have metadata operations performed on them.
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\Category_Colors
+ */
+class Event_Category_Meta {
+
+	/**
+	 * The taxonomy associated with event categories.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $taxonomy = Tribe__Events__Main::TAXONOMY;
+
+	/**
+	 * The ID of the term this instance operates on.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	protected int $term_id;
+
+	/**
+	 * Stores pending metadata updates before saving.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected array $pending_meta = [];
+
+	/**
+	 * Stores pending metadata deletions before saving.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected array $pending_deletes = [];
+
+	/**
+	 * Stores pending metadata updates before saving.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected array $pending_updates = [];
+
+	/**
+	 * Constructor.
+	 * Ensures the term belongs to `tribe_events_cat` before initializing.
+	 *
+	 * @since TBD
+	 * @throws InvalidArgumentException Thrown when an invalid term ID is provided or the term does not exist.
+	 *
+	 * @param int $term_id The term ID.
+	 */
+	public function __construct( int $term_id ) {
+		if ( $term_id <= 0 ) {
+			throw new InvalidArgumentException( __( 'Invalid term ID.', 'the-events-calendar' ) );
+		}
+
+		$term = get_term( $term_id, $this->taxonomy );
+
+		if ( ! $term instanceof WP_Term ) {
+			throw new InvalidArgumentException(
+			/* translators: %1$d is the term ID, %2$s is the taxonomy name. */
+				sprintf( __( 'Term ID %1$d does not exist in taxonomy %2$s.', 'the-events-calendar' ), $term_id, $this->taxonomy )
+			);
+		}
+
+		$this->term_id = $term_id;
+	}
+
+	/**
+	 * Sets metadata for the term.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key   The meta key to update.
+	 * @param mixed  $value The value to store.
+	 *
+	 * @return $this Returns the instance for chaining.
+	 */
+	public function set( string $key, $value ): self {
+		$key   = $this->validate_key( $key );
+		$value = $this->validate_value( $value );
+
+		// If key or value is invalid, skip but don't break chaining.
+		if ( is_wp_error( $key ) || is_wp_error( $value ) ) {
+			return $this;
+		}
```

Comment:
I think is better when fluent apis throw exception on errors.

Somehow you gotta know that something failed.